### DISCUSSION TITLE
Dilate Image over Alpha?

### DISCUSSION THREAD

**Discussion by Raildex** [dlemstra/Magick.NET#1448] [Help]
How do I dilate an image when I want to use the Alpha as a "mask"?
I want to achieve Padding, similar to this:
![Dilation](https://github.com/ImageMagick/ImageMagick/assets/14938934/ed88d96f-5a56-49b3-97f0-fdc61ff768bf)

I have tried

```csharp
newImage.Morphology(MorphologyMethod.Dilate, Kernel.Disk, Channels.Alpha, 4);
newImage.Morphology(MorphologyMethod.Dilate, Kernel.Disk, Channels.RGB, 4);
```
In the case of `Alpha`, it results in a black border around my image.
In the case of `RGB`, the image itself becomes very blurry.

Here is the image I want to dilate
![0](https://github.com/ImageMagick/ImageMagick/assets/14938934/fbf9e9bb-a3f5-405d-83e3-9051a810a117)

and this is the result after 
`newImage.Morphology(MorphologyMethod.Dilate, Kernel.Disk, Channels.Alpha);`
![0_](https://github.com/ImageMagick/ImageMagick/assets/14938934/12148859-3d08-45b0-85a9-a46c5a4937c9)

And here after I use `Channels.RGB`:
![0_](https://github.com/ImageMagick/ImageMagick/assets/14938934/21e98b81-ebfc-44d0-bf64-5fee72312120)

**Comment by dlemstra** [comment]
I will wait for the answer you are getting in the ImageMagick discussions. And then I can help you translate that to C# if you need help with that.
